### PR TITLE
Publish embassies index

### PR DESCRIPTION
**Pull request by chrisroos**
Trello link: https://trello.com/c/MMUcRbIT

This PR enables Whitehall to publish an `embassy_index` document to the Publishing API and keep that document up-to-date with changes made to its underlying data. We need this document to be published so that we can switch over the rendering of the `/world/embassies` page from Whitehall to Collections. 

---

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: lib/whitehall/publishing_app.rb
Comment (BeckaL) [inline lib/whitehall/publishing_app.rb:3]: Lovely little improvement, thank you ⭐

File: app/models/embassy.rb
Comment (BeckaL) [inline app/models/embassy.rb:24]: Just a check since you mention in the commit message about the template doing this transformation - does this affect the existing rendering (i.e. are we going to be calling `.name` in a template now on a string)?

File: app/models/contact.rb
Comment (BeckaL) [inline app/models/contact.rb:23]: Just leaving a comment here unrelated to the line but in response to the decisions you've outlined in this commit message that I agree with all the decisions you've made there, thanks for outlining them! 😄

**Review by BeckaL — COMMENTED** [review]
Looks great! Just one question here to double check that we're not affecting any existing rendering by changing something on the model which I imagine you've already thought about so just wanted to check, but happy to approve after that.

File: app/models/embassy.rb
Comment (chrislo) [reply app/models/embassy.rb:24]: Previously we were relying on the `to_s` method [defined by WorldLocation](https://github.com/alphagov/whitehall/blob/publish-embassies-index/app/models/world_location.rb#L61). So we've decided to call `name` explicitly here to populate the json. This doesn't affect the existing rendering.

File: app/models/contact.rb
Comment (chrislo) [reply app/models/contact.rb:23]: Thank you!

**Comment by chrislo** [comment]
@BeckaL - we've double checked the rendering question you asked, and we don't think there'll be an issue - let us know if you have any other thoughts.

**Review by BeckaL — APPROVED** [review]
Great, LGTM :shipit:

**Comment by chrislo** [comment]
Thanks @BeckaL! Rebasing and force-pushing before merging.

**Pull request merged**
